### PR TITLE
comparison operators

### PR DESCRIPTION
**Pull request by robinnedergard**

### CODE REVIEW

File: parse/lexer/Source.ts [hunk 1]
```diff
@@ -26,8 +26,11 @@ export class Source extends IO.BufferedReader implements Error.Handler {
 	until(endMark: string | string[]): Source {
 		return new Source(IO.UntilReader.create(this, endMark), this.errorHandler)
 	}
-	peekIsSymbol(): boolean {
-		return !!this.peekIs(["!", "(", ")", "[", "]", "|", "*", ":", "."])
+	peekIsDoubleSymbol(): boolean {
```
Comment (florianebmeier) [inline parse/lexer/Source.ts:29]: Remove peekIsDoubleSymbol

**Review by florianebmeier — CHANGES_REQUESTED** [review]

**Review by florianebmeier — DISMISSED** [review]

File: Exists.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { Rule } from "./Rule"
+
+export class Exists extends Rule {
```
Comment (simonmika) [inline Exists.ts:3]: I think `has` is a better name.

File: GreaterThan.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	readonly class = "GreaterThan"
+	constructor(readonly value: number | string) {
+		super()
+	}
+	is(value: any): boolean {
+		return typeof value == "number"
+			? typeof this.value == "string"
+				? value > Number.parseFloat(this.value)
+				: value > this.value
+			: value.toString().trim() > this.value.toString().trim()
```
Comment (simonmika) [inline GreaterThan.ts:14]: Flatten out.

File: GreaterThanOrEqual.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { Rule } from "./Rule"
+
+export class GreaterThanOrEqual extends Rule {
+	readonly precedence = Number.MAX_SAFE_INTEGER
+	readonly class = "GreaterThan"
+	constructor(readonly value: number | string) {
+		super()
+	}
+	is(value: any): boolean {
+		return typeof value == "number"
```
Comment (simonmika) [inline GreaterThanOrEqual.ts:10]: Flatten nesting.

File: LesserThan.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	readonly class = "LesserThan"
+	constructor(readonly value: number | string) {
+		super()
+	}
+	is(value: any): boolean {
+		return typeof value == "number"
+			? typeof this.value == "string"
+				? value < Number.parseFloat(this.value)
+				: value < this.value
+			: value.toString().trim() < this.value.toString().trim()
```
Comment (simonmika) [inline LesserThan.ts:14]: Flatten nesting.

File: LesserThanOrEqual.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+	readonly class = "LesserThan"
+	constructor(readonly value: number | string) {
+		super()
+	}
+	is(value: any): boolean {
+		return typeof value == "number"
+			? typeof this.value == "string"
+				? value <= Number.parseFloat(this.value)
+				: value <= this.value
+			: value.toString().trim() <= this.value.toString().trim()
```
Comment (simonmika) [inline LesserThanOrEqual.ts:14]: Flatten

**Review by simonmika — CHANGES_REQUESTED** [review]

**Review by simonmika — APPROVED** [review]

**Pull request merged**